database/postgres/endpoint.go

```diff
@@ -331,12 +369,19 @@ func (e *endpointRepo) CountProjectEndpoints(ctx context.Context, projectID stri
 }
 
 func (e *endpointRepo) FindEndpointByTargetURL(ctx context.Context, projectID string, targetURL string) (*datastore.Endpoint, error) {
-	endpoint := &datastore.Endpoint{}
-	err := e.db.QueryRowxContext(ctx, fetchEndpointByTargetURL, targetURL, projectID).StructScan(endpoint)
-	if err != nil {
-		if errors.Is(err, sql.ErrNoRows) {
-			return nil, datastore.ErrEndpointNotFound
+	endpoint, err := e.readFromCache(ctx, targetURL, func() (*datastore.Endpoint, error) {
```

Comment:
Will targetURL work here?

Reply:
@danvixent It should work, it's merely a cache key.

Reply:
@danvixent, It's doing lazy loading; after a miss, it retrieves it from the DB and saves it using the `targetURL` as the key.